src/libsystemd/sd-id128/id128-util.c

```diff
@@ -58,6 +61,10 @@ int id128_read_fd(int fd, Id128FormatFlag f, sd_id128_t *ret) {
          *     -ENOPKG:    "uninitialized" or "uninitialized\n",
          *     -EUCLEAN:   other invalid strings. */
 
+        fd = xopenat(dir_fd, path, O_RDONLY|O_CLOEXEC|O_NOCTTY, 0);
```

Comment:
I wonder if we should add `O_NOFOLLOW` here and require callers to chase() if they want to follow symlinks.